### PR TITLE
misc: add first/last_op to Block

### PR DESCRIPTION
**Pull request by superlopuh**
This is a part of the changes made for the doubly-linked list experiment. I think this API is useful enough to merge apart from the LL changes.

### CODE REVIEW

File: xdsl/ir.py [hunk 1]
```diff
@@ -972,6 +972,16 @@ def _attach_op(self, operation: Operation) -> None:
             )
         operation.parent = self
 
+    @property
+    def first_op(self) -> Operation | None:
+        """The first operation in `self`"""
```
Comment (math-fehr) [inline xdsl/ir.py:977]:
```suggestion
        """The first operation in the block."""
```
Maybe ?

File: xdsl/ir.py [hunk 2]
```diff
@@ -972,6 +972,16 @@ def _attach_op(self, operation: Operation) -> None:
...
+        return self.ops[0] if len(self.ops) else None
+
+    @property
+    def last_op(self) -> Operation | None:
+        """The last operation in `self`"""
```
Comment (math-fehr) [inline xdsl/ir.py:982]:
```suggestion
        """The last operation in the block."""
```
Maybe?

**Review by math-fehr — APPROVED** [review]
Really nice! That should make things easier to move to linked lists!

File: xdsl/ir.py (hunk 2 above)
Comment (superlopuh) [reply xdsl/ir.py:982]: How about `"""The last operation in this block."""`?

**Review by webmiche — APPROVED** [review]
It seems this functions are indeed useful, so LGTM.

**Pull request merged**